### ISSUE TITLE
Intermittent Travis CI Issues

### ISSUE THREAD

**Issue by jonathan-m-hamilton** [hyperledger-labs/blockchain-integration-framework#36]
Investigate flaky Travis CI failures and determine if one or more fixes required

**Comment by jonathan-m-hamilton** [comment]
Example 1
`> @hyperledger/blockchain-integration-framework@0.1.0 test /home/travis/build/hyperledger-labs/blockchain-integration-framework
403> cross-env NODE_ENV=test ./node_modules/.bin/mocha --ui bdd --reporter spec --colors tests/*.js --recursive --exit
404
405/home/travis/build/hyperledger-labs/blockchain-integration-framework/node_modules/mocha/node_modules/yargs/yargs.js:1163
406 else throw err
407 ^
408
409Error: Cannot find module 'typedarray-to-buffer'
410 at Function.Module._resolveFilename (module.js:548:15)
411 at Function.Module._load (module.js:475:25)
412 at Module.require (module.js:597:17)
413 at require (internal/module.js:11:18)
414 at Object.<anonymous> (/home/travis/build/hyperledger-labs/blockchain-integration-framework/node_modules/web3-providers-ws/node_modules/websocket/lib/W3CWebSocket.js:18:16)
415 at Module._compile (module.js:653:30)
416 at Object.Module._extensions..js (module.js:664:10)
417 at Module.load (module.js:566:32)
`

**Comment by jonathan-m-hamilton** [comment]
Example 2 https://travis-ci.com/hyperledger-labs/blockchain-integration-framework/builds/136396597

**Comment by jonathan-m-hamilton** [comment]
Hopefully not as flaky as thought 😎

`
/home/travis/build/hyperledger-labs/blockchain-integration-framework/node_modules/mocha/node_modules/yargs/yargs.js:1163
400 else throw err
401 ^
402
403Error: Cannot find module 'typedarray-to-buffer'
`

**Comment by petermetz** [comment]
Clue: npm install and npm test both succeed on local ubuntu VM, but the missing dependency is highlighted as an issue there as well.

Some code in a dependency might be importing different packages at runtime based on some criteria that differentiates between the Travis CI VM and the local VM I'm testing with:

```sh
$ npm ls typedarray-to-buffer
@hyperledger-labs/blockchain-integration-framework@0.1.0 /home/multipass/a/blockchain/blockchain-integration-framework
└─┬ web3@1.0.0-beta.37
  └─┬ web3-core@1.0.0-beta.37
    └─┬ web3-core-requestmanager@1.0.0-beta.37
      └─┬ web3-providers-ws@1.0.0-beta.37
        └─┬ websocket@1.0.26 invalid (git://github.com/frozeman/WebSocket-Node.git#6c72925e3f8aaaea8dc8450f97627e85263999f2)
          └── UNMET DEPENDENCY typedarray-to-buffer@^3.1.2

npm ERR! invalid: websocket@1.0.26 /home/multipass/a/blockchain/blockchain-integration-framework/node_modules/web3-providers-ws/node_modules/websocket
npm ERR! missing: typedarray-to-buffer@^3.1.2, required by websocket@1.0.26
```

**Comment by petermetz** [comment]
Tried upgrading web3 to latest stable but that has another issue that breaks dependencies (`v1.2.3`)
https://github.com/ethereum/web3.js/issues/3210

Will try with web3 `1.2.2`

**Comment by petermetz** [comment]
I tinkered with the travis build, had to update the web3 version from that beta release to a stable one and put in some shell code to ensure npm dependencies are installed in the example sub-projects as well (fabric/api, quorum/api)

There is still the issue of flakiness which is a different one, it appears to be in one of the unit tests, but that doesn't always fail, just sometimes.

For now I put the fixed up code here in a draft PR, will hopefully come up with something for the flaky unit test tomorrow:
#50

**Comment by petermetz** [comment]
Web3 released a fix for the dependency issues in `v1.2.3` this morning, going to use that:
https://github.com/ethereum/web3.js/releases
